server/service/osquery_utils/queries.go

```diff
@@ -1566,6 +1572,13 @@ func directIngestMacOSProfiles(
 	return apple_mdm.VerifyHostMDMProfiles(ctx, ds, host, installed)
 }
 
+func directIngestMDMDeviceIDWindows(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error {
+	if len(rows) != 1 {
```

Comment:
TODO: Confirm the expected behavior if the registry key is not present on the device. Should we ignore `len(rows) == 0` or return error?

Reply:
FYI - The registry key is only going to be present if the device is MDM-enrolled. Should we return empty instead?

Reply:
@marcosd4h, would you expect the key to persist when a device is enrolled to MDM via fleetd and then disconnected from MDM via the "Access work or school" setting on the device? I tried this now and after the device is disconnected, I'm still able to live query for the device ID.

Reply:
yes, that's expected. Once set, the DeviceID will change only after a factory reset or major OS upgrade. The next MDM enrollment will use the same DeviceID (you can try this by disconnecting MDM via "Access work or school", and then 
waiting for fleetd to re-enroll the device - DeviceID should be the same)

Reply:
Thanks, that's what I saw when re-enrolling!

Reply:
@marcosd4h, do we also expect that a host that was disconnected via "Access work or school" will reconnect to MDM in Fleet whenever the device is restarted (assuming that Windows MDM is still enabled on the Fleet server)?